include/PathUtil.h

```diff
@@ -0,0 +1,29 @@
+#ifndef PATHUTIL_H
+#define PATHUTIL_H
+
+#include "lmms_export.h"
+
+#include <QDir>
+
+namespace PathUtil
+{
+	enum Base { AbsoluteBase, ProjectDirBase, FactorySampleBase, UserSampleBase, UserVSTBase, PresetBase,
+		UserLADSPABase, DefaultLADSPABase, UserSoundfontBase, DefaultSoundfontBase, UserGIGBase, DefaultGIGBase };
+
+	QString LMMS_EXPORT baseLocation(const Base & base);
```

Comment:
`Base` doesn't need passing by reference - it's no more expensive to pass than the reference itself, and the extra indirection can add overhead.